mint-choc-chip-skyblade/sshd-rando
# Boss Key Puzzles

## What does this PR do?
Adds an option to change the behavior of boss key puzzles. The available options are `correct_orientation`, `vanilla_orientation`, and `random_orientation`. `correct_orientation` will orient each boss key so that it can be inserted immediately, `vanilla_orientation` doesn't change anything, and `random_orientation` will randomly change each boss key's initial insert angle.

I added an extra field `RNG_SEED` which is generated during game patching and gives everybody who is generating the same randomizer the same random random boss key orientations. 

I also added proper support for adding dungeon items to the starting inventory (maps, small keys, boss keys). 

## How do you test this changes?
I went to each boss key to verify that each one's correct position was valid (conveniently they all have the same correct position). I also tested starting with all the dungeon items and went to each dungeon to verify that I had all of them.
